railties/test/application/configuration_test.rb

```diff
@@ -2398,6 +2398,24 @@ def index
       assert_equal :default, Rails.configuration.debug_exception_response_format
     end
 
+    test "ActiveRecord configs are applied from initializers even loaded prior to initializers running" do
+      # Set to defaults for 6.1 to default to verify_foreign_keys_for_fixtures being false
+      # and establish a DB connection in config to trigger ActiveRecord loading before initializers run
+      remove_from_config '.*config\.load_defaults.*\n'
+      add_to_config <<-RUBY
+        config.load_defaults "6.1"
+        ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+      RUBY
```

Comment:
I think this is what's causing the issue for you. You shouldn't be loading Active Record during initialization at all

Reply:
Yes, I'm very aware this shouldn't be happening. The actual issue for me was that a gem was causing ActiveRecord to load when the gem was required.

This is more trying to make things work a bit better even when a thing that shouldn't happen happens.

Reply:
Could you share what the gem is? I agree that Rails should handle this (others may not) but regardless it may be worth raising an issue with the gem.

Reply:
Someone else had already opened a similar issue with the gem here: https://github.com/ilyakatz/data-migrate/issues/222. So I added some info to that as well, and my goal is that the gem will also stop having this behavior.

This is definitely doing a bad thing with ActiveRecord, but I do think the change in this PR makes things work better even when you do something that you shouldn't do.